ssl/statem/statem_lib.c

```diff
@@ -704,6 +704,44 @@ static int version_cmp(const SSL *s, int a, int b)
     return DTLS_VERSION_LT(a, b) ? -1 : 1;
 }
 
+int ssl_strict_version_check(const SSL *s, int version)
+{
+    int mask;
+
+    switch (version) {
+    case SSL2_VERSION:
+      mask = SSL_OP_NO_SSLv2;
+    case SSL3_VERSION:
+      mask = SSL_OP_NO_SSLv3;
+    case TLS1_VERSION:
+      mask = SSL_OP_NO_TLSv1;
+      break;
+    case TLS1_1_VERSION:
+      mask = SSL_OP_NO_TLSv1_1;
+      break;
+    case TLS1_2_VERSION:
+      mask = SSL_OP_NO_TLSv1_2;
+      break;
+    default:
+      mask = 0;
+      break;
+    }
+
+    /* Version should not be disabled by SSL_set_options */
+    if ((s->options & mask) != 0)
+        return 0;
+
```

Comment:
I thought about just going over (d)tls_version_table and checking that the version is enabled like ssl_get_client_min_max_version() does, and stop when the version from the table gets lower than the one from the session.